modules/react/button/lib/BaseButton.tsx

```diff
@@ -292,13 +126,265 @@ export const getPaddingStyles = (
+      fill: cssVar(buttonVars.hover.icon, cssVar(base.blackPepper500)),
+    },
+  },
+  '&:hover:active': {transitionDuration: '40ms'},
+  '&:active, &.active': {
+    backgroundColor: cssVar(buttonVars.active.background, 'transparent'),
+    borderColor: cssVar(buttonVars.active.border, 'transparent'),
+    color: cssVar(buttonVars.active.label, cssVar(base.blackPepper400)),
+    '& span .wd-icon-fill': {
+      fill: cssVar(buttonVars.active.icon, cssVar(base.blackPepper400)),
+    },
+    '.wd-icon-background ~ .wd-icon-accent, .wd-icon-background ~ .wd-icon-accent2': {
+      fill: cssVar(buttonVars.active.icon),
+    },
+  },
+  '&:disabled, &:active:disabled, &:focus:disabled, &:hover:disabled': {
+    backgroundColor: cssVar(buttonVars.disabled.background, 'transparent'),
+    borderColor: cssVar(buttonVars.disabled.border, 'transparent'),
+    color: cssVar(buttonVars.disabled.label, cssVar(base.blackPepper400)),
+    '& span .wd-icon-fill': {
+      fill: cssVar(buttonVars.disabled.icon, cssVar(base.blackPepper400)),
+    },
+    '.wd-icon-background ~ .wd-icon-accent, .wd-icon-background ~ .wd-icon-accent2': {
+      fill: cssVar(buttonVars.disabled.icon, cssVar(base.blackPepper400)),
+    },
+  },
+});
+
+/**
+ * Button modifiers that will overwrite the base styles of Buttons.
+ * - `Size`: These modifiers will dictate a size of a Button and has a set of styles to associated with it.
+ * - `iconPosition`: These modifiers will override the existing `Size` styles. These are specific to icon locations
+ * within a button or if there is only an icon and no text.
+ */
+export const buttonModifiers = createModifiers({
+  size: {
+    large: createStyles({
+      fontSize: cssVar(system.space.x4, space.s),
+      lineHeight: cssVar(system.space.x6, space.m),
+      letterSpacing: '0.01rem',
+      height: '48px',
+      paddingInline: cssVar(system.space.x8, space.l),
+      minWidth: '112px',
+    }),
+    medium: createStyles({
+      fontSize: '0.875rem',
+      letterSpacing: '0.015rem',
+      minWidth: '96px',
+      paddingInline: cssVar(system.space.x6, space.m),
+      height: cssVar(system.space.x10, space.xl),
+    }),
+    small: createStyles({
+      fontSize: '0.875rem',
+      letterSpacing: '0.015rem',
+      height: cssVar(system.space.x8, space.l),
+      minWidth: cssVar(system.space.x20, space.xxxl),
+      paddingInline: cssVar(system.space.x4, space.s),
+      gap: cssVar(system.space.x1, space.xxxs),
+    }),
+    extraSmall: createStyles({
+      fontSize: '0.75rem',
+      lineHeight: cssVar(system.space.x4, space.s),
+      letterSpacing: '0.02rem',
+      height: cssVar(system.space.x6, space.m),
+      minWidth: 'auto',
+      paddingInline: cssVar(system.space.x3, space.xs),
+      gap: cssVar(system.space.x1, space.xxxs),
+    }),
+  },
+  iconPosition: {
+    largeOnly: createStyles({
+      padding: '0',
+      minWidth: cssVar(system.space.x12, '3rem'),
```

Comment:
I'd like to get rid of all fallbacks to system variables. The optimizer will be able to parse the CSS files from our tokens package and automatically fill in fallbacks. This would be a win for us in Canvas Kit because we don't have to lookup fallbacks manually, but still get the benefit of working in environments where the variables are not provided (most likely Storybook)